core-common/src/main/java/org/zalando/nakadi/annotations/validation/DataLakeAnnotationValidator.java

```diff
@@ -7,29 +7,47 @@
+            "^(unlimited|(([7-9]|[1-9]\\d{1,2}|[1-2]\\d{3}|3[0-5]\\d{2}|36[0-4]\\d|3650)((\\sdays?)|(d)))" +
+                    "|(([1-9][0-9]?|[1-4][0-9]{2}|5([0-1][0-9]|2[0-1]))((\\sweeks?)|(w)))|" +
+                    "(([1-9]|[1-9]\\d|[1][01]\\d|120)((\\smonths?)|(m)))|(([1-9]|(10))((\\syears?)|(y))))$");
     public static final String RETENTION_PERIOD_ANNOTATION = "datalake.zalando.org/retention-period";
     public static final String RETENTION_REASON_ANNOTATION = "datalake.zalando.org/retention-period-reason";
+    public static final String MATERIALISE_EVENTS_ANNOTATION = "datalake.zalando.org/materialize-events";
 
     @Override
     public boolean isValid(final Map<String, String> annotations, final ConstraintValidatorContext context) {
         if (annotations == null || annotations.size() == 0) {
             return true;
         }
+        if (annotations.containsKey(MATERIALISE_EVENTS_ANNOTATION)) {
+            if (!annotations.get(MATERIALISE_EVENTS_ANNOTATION).equals("off") &&
+                    !annotations.get(MATERIALISE_EVENTS_ANNOTATION).equals("on")) {
+                context.disableDefaultConstraintViolation();
+                context.buildConstraintViolationWithTemplate("Field " + MATERIALISE_EVENTS_ANNOTATION
+                                + " is not valid. Provided value: \""
+                                + annotations.get(MATERIALISE_EVENTS_ANNOTATION)
+                                + "\". Possible values are: \"on\" or \"off\".")
+                        .addConstraintViolation();
+                return false;
+            }
+        }
+
         if (annotations.containsKey(RETENTION_PERIOD_ANNOTATION)) {
             if (annotations.getOrDefault(RETENTION_REASON_ANNOTATION, "").equals("")) {
                 context.disableDefaultConstraintViolation();
                 context.buildConstraintViolationWithTemplate(
-                                "Retention reason is required, when " + RETENTION_PERIOD_ANNOTATION + " is specified")
+                                "Field " + RETENTION_REASON_ANNOTATION + " is required, when "
+                                        + RETENTION_PERIOD_ANNOTATION + " is specified.")
                         .addConstraintViolation();
                 return false;
             }
 
             if (!ANNOTATIONS_PERIOD_PATTERN.matcher(annotations.get(RETENTION_PERIOD_ANNOTATION)).find()) {
                 context.disableDefaultConstraintViolation();
-                context.buildConstraintViolationWithTemplate(RETENTION_PERIOD_ANNOTATION +
-                                " does not comply with pattern: " + ANNOTATIONS_PERIOD_PATTERN)
+                context.buildConstraintViolationWithTemplate("Field " + RETENTION_PERIOD_ANNOTATION +
+                                " does not comply with regex. See documentation " +
+                                "(https://docs.google.com/document/d/1-SwwpwUqauc_pXu-743YA1gO8l5_R_Gf4nb" +
```

Comment:
Lets have the error message loud and clear. Documents might be updated, deleted or stale and it requires extra effort to keep the code and the document up to date.

Reply:
I understand your concerns here and I agree that it requires extra effort, but we did it intentionally, because we would like to provide quite a lot of context to our users. We believe that they should keep this documentation in mind whenever they work with this feature and we plan to change this documentation/guideline quite frequently, I would prefer to avoid changing the error in the code every time we do that.